Artifacts/windows-add-local-admin/artifact.ps1

```diff
@@ -0,0 +1,137 @@
+function Get-LocalGroupMembers ([Parameter(Mandatory = $true)] [string] $GroupName)
+{
+    if ($PSVersionTable.PSVersion.Major -lt 5)
+    {
+        return Get-LocalGroupMembersPs3to5 $GroupName
+    }
+
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group");
+    $members = @($group.psbase.Invoke("Members"));
+    $Details = @();
+    
+    $members | ForEach-Object {
+        $name = $_.GetType.Invoke().InvokeMember("Name", 'GetProperty', $null, $_, $null)
+        $path = $_.GetType.Invoke().InvokeMember("ADsPath", 'GetProperty', $null, $_, $null)
+        $schema = $_.GetType.Invoke().InvokeMember("Schema", 'GetProperty', $null, $_, $null)
+
+        if ($schema -notlike "WinNT://$name/*")
+        {
+            $Type = "Domain"
+            if ($path -like "*/$env:COMPUTERNAME/*")
+            {
+                $Type = "Local"
+            }
+           
+            $Details += New-Object PSObject -Property @{"Group" = $GroupName; "Account" = $name; "Type" = $type; }
+        }
+    }
+    Write-Output "Members of $GroupName are:"
+    $Details.ForEach( {$_}) | Format-Table -Property Account, Type -AutoSize
+}
+    
+##############################
+# Function to get the AD User string in format "WinNT://<domain>/<username>"
+function Get-AdUsernameString ([Parameter(Mandatory = $true)] [string] $UserName)
+{
+    if ([string]::IsNullOrEmpty($Username)) 
+    { 
+        Write-Error "Username not provided"
+        throw "Username not provided"
+    }
+
+    if ($Username -notmatch '\\')
+    {
+        $ADResolved = ($Username -split '@')[0]
+        $DomainResolved = ($Username -split '@')[1]    
+        $Username = 'WinNT://', "$DomainResolved", '/', $ADResolved -join ''
+    }
+    else
+    {
+        $ADResolved = ($Username -split '\\')[1]
+        $DomainResolved = ($Username -split '\\')[0]
+        $Username = 'WinNT://', $DomainResolved, '/', $ADResolved -join ''
+    }
+    return $Username
+}
+
+##############################
+# Function to add an AD user to a local group
+function Add-UserToLocalGroup ([Parameter(Mandatory = $true)] [string] $Username, [Parameter(Mandatory = $true)] [string] $GroupName)
+{
+    Write-Output "Attempting to add $DomainJoinUsername to the administrators group..."
+    $adUser = Get-AdUsernameString $Username
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group")
+
+    if ($group.IsMember($adUser))
+    {
+        Write-Output "Result: $Username already belongs to the $GroupName"
+        return
+    }
+    $group.Add($adUser)
+    
+    if ($group.IsMember($adUser))
+    {
+        Write-Output "Result: $Username successfully added to $GroupName"
+    }
+    else
+    {
+        Write-Error "Result: failed to add $username to $GroupName"
+    }
+}
+
+##############################
+# Main function
+$adminGroupName = Get-LocalAdminGroupName
+Add-UserToLocalGroup -Username $DomainJoinUsername -GroupName $adminGroupName
+Get-LocalGroupMembers $adminGroupName
```

Comment:
Add `Write-Output "Members of $GroupName are:"` a line before calling `Get-LocalGroupMembers $adminGroupName`.